### PR TITLE
Fix dragger on touch devices

### LINKED ISSUE #162: Dragger often doesn't work for touch* events [patch]
This issue was imported from Closure Library's previous home at http://closure-library.googlecode.com

The link to the original issue is:
https://code.google.com/p/closure-library/issues/detail?id=382

### PR DESCRIPTION
**Pull request by elemoine**
Closes #162

### CODE REVIEW

**Comment by elemoine** [comment]
Here is an example that uses a patched version of `goog.fx.Dragger`: http://erilem.net/ngeo/dragdrop/layerorder.html. `goog.fx.DragListGroup` is used really in this example.

**Comment by elemoine** [comment]
Any chance to see this merged? Is there anything else I can do?

**Comment by MatrixFrog** [comment]
I'll see if I can find someone who knows more about touch events than I do. It would probably also help if the commit description was more detailed: Fix it how? What was broken about it? etc.

**Comment by elemoine** [comment]
@MatrixFrog, thanks. I rebased the commit onto the latest master, and I added a more detailed description of the problem and of the solution in the commit message. I hope this helps.

**Comment by elemoine** [comment]
@MatrixFrog, out of curiosity,  did you finally have someone else review this?

**Comment by MatrixFrog** [comment]
No, I ended up reviewing it myself. I'm not an expert on this stuff but it seems pretty clear that you know what you're doing. Having a good commit description is very helpful :)

Sorry for the delay, thanks for your patience.

**Comment by elemoine** [comment]
No problem, I know what it is.

**Comment by nanaze** [comment]
long ago goog.labs.events.touch was introduced to merge our touch/pointer handling. I would hope that we could start solidifying on that.

**Comment by mtsgrd** [comment]
@elemoine @MatrixFrog I don't have hard evidence for you, but I've just tested this for my closure project and dragging now works on iOS 7, but not on iOS 8. Any thoughts?

**Comment by mtsgrd** [comment]
This is probably related: http://developer.telerik.com/featured/scroll-event-change-ios-8-big-deal/

**Comment by elemoine** [comment]
> long ago goog.labs.events.touch was introduced to merge our touch/pointer handling. I would hope that we could start solidifying on that.

I agree that a better event architecture integrating touch/pointer handling would be a good thing.

**Comment by fredj** [comment]
@mtsgrd is there any error messages from Safari? 
I suspect https://bugs.dojotoolkit.org/ticket/18168#comment:4

**Comment by elemoine** [comment]
Thanks @fredj. I'll have a look.

File: closure/goog/events/browserevent.js [hunk 1]
```diff
@@ -223,8 +223,19 @@ goog.events.BrowserEvent.IEButtonMap = [
 goog.events.BrowserEvent.prototype.init = function(e, opt_currentTarget) {
   var type = this.type = e.type;
 
+  /** @type {Touch} */
+  var relevantTouch = null;
+  if (type == goog.events.EventType.TOUCHSTART ||
+      type == goog.events.EventType.TOUCHMOVE) {
+    relevantTouch = e.targetTouches[0];
```
Comment (elemoine) [inline closure/goog/events/browserevent.js]: After more thinking, and looking at #337, I think it's a mistake to rely on `targetTouches`. With multiple touches on the touch surface `targetTouches[0]` is unlikely to be the "relevant touch". See my comments in #337 for more details, and see the `targetTouches` definition on [mdn](https://developer.mozilla.org/en-US/docs/Web/API/TouchEvent/targetTouches). So as done in #337 I'm tempted to rewrite this whole block as follows:

```js
  /**
   * On touch devices use the first "changed touch" as the relevant touch.
   * @type {Touch}
   */
  var relevantTouch = e.changedTouches ? e.changedTouches[0] : null;
```

Comment (elemoine) [inline closure/goog/events/browserevent.js]: Here's a branch implementing this: https://github.com/elemoine/closure-library/tree/targettouches. @mtsgrd, I'd be curious to know if that makes a difference with iOS 8.

Comment (elemoine) [inline closure/goog/events/browserevent.js]: @mtsgrd this example is based on my `targettouches` branch: http://erilem.net/ngeo/layerorder-touch/layerorder.html. Does it work for you in iOS 8?

**Pull request merged**